tests/test_buffer.py

```diff
@@ -666,6 +666,42 @@ def test_SimpleArray_broadcast_slice_ghost_md(self):
                 for k in range(4):
                     self.assertEqual(ndarr2[i, j, k], sarr[i - G, j, k])
 
+    def test_SimpleArray_broadcast_from_list_list(self):
```

Comment:
Is the testing code self-explaining?

Reply:
It's pretty good.  You can use 3 test functions or combine them into one.  Both are OK.  Sometimes the latter is easier to maintain.